Lucid/Core/APIClient.swift

```diff
@@ -599,6 +632,72 @@ extension APIClient {
         }
     }
 
+    public func send(request: APIRequest<Data>) async -> Result<APIClientResponse<Data>, APIError> {
+        let requestConfig = await prepareRequest(request.config)
+
+        let requestDescription = self.description(for: requestConfig)
+
+        if await self.deduplicator.isDuplicated(request: requestConfig) {
+            return await self.deduplicator.waitForDuplicated(request: requestConfig)
```

Comment:
Just curious, would this would better as a guard to de-scope the rest of the code? 

I'm totally fine with this as is.

Reply:
I think it will look even better just removing the `else`. Thoughts?